guvnor-m2repo-editor/guvnor-m2repo-editor-backend/src/main/java/org/guvnor/m2repo/backend/server/GuvnorM2Repository.java

```diff
@@ -509,11 +513,9 @@ private RemoteRepository getRemoteRepoFromDeployment( DeploymentRepository repo
         Server server = settings.getServer( repo.getId() );
 
         if ( server != null ) {
-            remoteRepoBuilder.setAuthentication(
-                    new AuthenticationBuilder()
-                            .addUsername( server.getUsername() )
-                            .addPassword( server.getPassword() )
-                            .build() );
+            Authentication authentication = embedder.getMavenSession().getRepositorySession()
+                    .getAuthenticationSelector().getAuthentication( remoteRepoBuilder.build() );
```

Comment:
@afcollins Thanks for the explanation and link. When @mariofusco has confirmed he's happy with the other (Drools) PR I can request this be merged into master too - I might fork your fork and try integrating your test into the existing module (IDK what I can do that you have not tried, but I'd prefer the test in the same module as the change - and using the existing Aether.. if possible).